### ISSUE TITLE
99

### ISSUE THREAD

**Issue by bryanjohnabraham** [bryanjohnabraham/ml#12]
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier, plot_tree
from sklearn.metrics import accuracy_score, classification_report

df = pd.read_csv("Online Retail.csv", encoding="latin1")

df = df.dropna(subset=["CustomerID"])
df = df[df["Quantity"] > 0]
df = df[df["UnitPrice"] > 0]

df["TotalPrice"] = df["Quantity"] * df["UnitPrice"]

customer = df.groupby("CustomerID").agg({
    "Quantity": "sum",
    "TotalPrice": "sum",
    "InvoiceNo": "nunique"
}).reset_index()

customer.columns = [
    "CustomerID",
    "TotalQuantity",
    "TotalSpent",
    "NumPurchases"
]

customer["Segment"] = pd.qcut(
    customer["TotalSpent"],
    q=3,
    labels=["Low", "Medium", "High"]
)

X = customer[["TotalQuantity", "NumPurchases"]]
y = customer["Segment"]

X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.2,
    random_state=42,
    stratify=y
)

model = DecisionTreeClassifier(
    criterion="entropy",
    max_depth=4,
    random_state=42
)

model.fit(X_train, y_train)

y_pred = model.predict(X_test)

print("Decision Tree using ID3")
print("-----------------------")
print("Accuracy:", accuracy_score(y_test, y_pred))

print("\nClassification Report:")
print(classification_report(y_test, y_pred))

print("\nFeature Importance:")
for feature, importance in zip(X.columns, model.feature_importances_):
    print(feature, ":", importance)

plt.figure(figsize=(14, 8))

plot_tree(
    model,
    feature_names=X.columns,
    class_names=["Low", "Medium", "High"],
    filled=True
)

plt.title("Decision Tree using ID3")
plt.show()
